ultralytics/yolo/data/dataloaders/stream_loaders.py

```diff
@@ -98,8 +98,20 @@ def __next__(self):
             cv2.destroyAllWindows()
             raise StopIteration
 
-        im0 = self.imgs.copy()
-        return self.sources, im0, None, ''
+        # Wait for an image to be available
+        while all(img is None for img in self.imgs):
+            time.sleep(1 / max(self.fps))
+            if not all(x.is_alive() for x in self.threads) or cv2.waitKey(1) == ord('q'):  # q to quit
+                cv2.destroyAllWindows()
+                raise StopIteration
+
+        # Reset self.imgs and return available images
+        self.imgs, im0 = [None] * len(self.imgs), self.imgs
+        available = [i for i, img in enumerate(im0) if img is not None]
```

Comment:
We can not return anything other than an identically shaped output for every frame. If multiple streams are present and some are lagging then we can return a black frame for these, but understandably every frame in the dataloader needs to be the same shape. The `available` variable here violates this.